src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -5117,6 +5117,23 @@ class spell_gen_ancestral_call : public SpellScript
     }
 };
 
+// 83781 - Reverse Cast Ride Vehicle
+class spell_gen_reverse_cast_ride_vehicle : public SpellScript
+{
+    PrepareSpellScript(spell_gen_reverse_cast_ride_vehicle);
+
+    void HandleScript(SpellEffIndex effIndex)
+    {
+        if (Unit* caster = GetCaster())
+            GetHitUnit()->CastSpell(caster, GetSpellInfo()->GetEffect(effIndex).BasePoints, true);
```

Comment:
Better to name this reverse_cast_target_to_caster_triggered (reverse_cast_target_to_caster for non-triggered, I guess) or simply reverse_cast, this spell script can be used by hundreds spells
Basepoints should be calculated first and caster here cannot be null

Reply:
what do you mean by calculating basepoints exactly? can you explain it to me?

Reply:
Yeah